src/rebar_compiler_epp.erl

```diff
@@ -37,6 +47,120 @@ resolve_module(Mod, Paths) ->
         Path -> {ok, Path}
     end.
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%% Cache for deps      %%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%
+-spec ensure_started() -> ok.
+ensure_started() ->
+    case whereis(?MODULE) of
+        undefined ->
+            case gen_server:start({local, ?MODULE}, ?MODULE, [], []) of
+                {ok, _Pid} ->
+                    ok;
+                {error, {already_started, _Pid}} ->
+                    ok
+            end;
+        Pid when is_pid(Pid) ->
+            ok
+    end.
+
+flush() ->
+    gen_server:cast(?MODULE, flush).
+
+%% @doc Resolves "Name" erl module to a path, given list of paths to search.
+%% Caches result for subsequent requests.
+-spec resolve_source(atom() | file:filename_all(), [file:filename_all()]) -> {true, file:filename_all()} | false.
+resolve_source(Name, Dirs) when is_atom(Name) ->
+    gen_server:call(?MODULE, {resolve, atom_to_list(Name) ++ ".erl", Dirs});
+resolve_source(Name, Dirs) when is_list(Name) ->
+    gen_server:call(?MODULE, {resolve, Name, Dirs}).
+
+-record(state, {
+    %% filesystem cache, denormalised
+    fs = #{} :: #{file:filename_all() => [file:filename_all()]},
+    %% map of module name => abs path
+    resolved = #{} :: #{file:filename_all() => file:filename_all()}
+}).
+
+init([]) ->
+    {ok, #state{}}.
+
+handle_call({resolve, Name, Dirs}, _From, #state{fs = Fs, resolved = Res} = State) ->
+    case maps:find(Name, Res) of
+        {ok, Found} ->
+            {reply, Found, State};
+        error ->
+            {Resolved, NewFs} = resolve(Name, Fs, Dirs),
+            {reply, Resolved, State#state{resolved = Res#{Name => Resolved}, fs = NewFs}}
+    end.
+
+handle_cast(flush, _State) ->
+    {noreply, #state{}}.
+
+resolve(_Name, Fs, []) ->
+    {false, Fs};
+resolve(Name, Fs, [Dir | Tail]) ->
+    {NewFs, Files} = list_directory(Dir, Fs),
+    case lists:member(Name, Files) of
+        true ->
+            {{true, filename:join(Dir, Name)}, NewFs};
+        false ->
+            resolve(Name, NewFs, Tail)
+    end.
```

Comment:
Currently writing a fix for this; this search needs to be recursive in order to work since we support directory recursion. Not having it is currently causing issues for some builds in 3.14, but I'll try to re-shape it such that it doesn't cause problems in terms of performance if you don't _need_ recursion to work.

Reply:
Seems there's already some recursion below, so there might just be an issue with properly grabbing sub-paths.